src/libraries/System.IO.FileSystem.AccessControl/src/System/IO/MemoryMappedFiles/MemoryMappedFileAcl.cs

```diff
@@ -0,0 +1,434 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.MemoryMappedFiles
+{
+    public static class MemoryMappedFileAcl
```

Comment:
The `CreateFromFile`, `CreateNew` and `CreateOrOpen` methods are almost identical to the ones that already exist in the `MemoryMappedFile` class. They only differ in the section that creates the file handle with the provided ACL configuration, which uses the two constructors that had to be made public.

Reply:
To avoid requesting to make the `MemoryMappedFile` constructor that receives 3 arguments public, I am calling the one that was approved, and indicating in the documentation that the `leaveOpen` argument is unused: the handle will always be left open after the memory-mapped file is disposed.

Reply:
Update: I asked @bartonjs for his advice on this, and the other constructor can be approved if the second argument is not a file stream, but an `IDisposable` (since the only thing we need from that object is its `Dispose()` method).
The code has been updated to reflect this, and will bring the API to review on the next meeting.